packages/ai-hugging-face/src/node/huggingface-language-model.ts

```diff
@@ -0,0 +1,129 @@
+}
+
+function toRoleLabel(actor: MessageActor): string {
+    switch (actor) {
+        case 'user':
+            return 'User';
+        case 'ai':
+            return 'Assistant';
+        case 'system':
+            return 'System';
+        default:
+            return '';
+    }
+}
+
+export class HuggingFaceModel implements LanguageModel {
+
+    /**
+     * @param id the unique id for this language model. It will be used to identify the model in the UI.
+     * @param model the model id as it is used by the Hugging Face API
+     * @param apiKey function to retrieve the API key for Hugging Face
+     */
+    constructor(public readonly id: string, public model: string, public apiKey: () => string | undefined) {
+    }
+
+    async request(request: LanguageModelRequest, cancellationToken?: CancellationToken): Promise<LanguageModelResponse> {
+        const hfInference = this.initializeHfInference();
+        if (this.isStreamingSupported(this.model)) {
+            return this.handleStreamingRequest(hfInference, request, cancellationToken);
+        } else {
+            return this.handleNonStreamingRequest(hfInference, request);
+        }
+    }
+
+    protected async handleNonStreamingRequest(hfInference: HfInference, request: LanguageModelRequest): Promise<LanguageModelTextResponse> {
+        const response = await hfInference.textGeneration({
+            model: this.model,
+            inputs: toHuggingFacePrompt(request.messages),
+            parameters: {
+                temperature: 0.1,          // Controls randomness, 0.1 for consistent outputs
+                max_new_tokens: 200,       // Limits response length
+                return_full_text: false,   // Ensures only the generated part is returned, not the prompt
+                do_sample: true,           // Enables sampling for more varied responses
+                stop: ['<|endoftext|>']    // Stop generation at this token
+            }
+        });
+
+        const cleanText = response.generated_text.replace(/<\|endoftext\|>/g, '');
+
+        return {
+            text: cleanText
+        };
+    }
+
+    protected async handleStreamingRequest(hfInference: HfInference, request: LanguageModelRequest, cancellationToken?: CancellationToken): Promise<LanguageModelResponse> {
+        const stream = hfInference.textGenerationStream({
+            model: this.model,
+            inputs: toHuggingFacePrompt(request.messages),
+            parameters: {
+                temperature: 0.1,
+                max_new_tokens: 200,
+                return_full_text: false,
+                do_sample: true,
+                stop: ['<|endoftext|>']
+            }
+        });
+
+        const asyncIterator = {
+            async *[Symbol.asyncIterator](): AsyncIterator<LanguageModelStreamResponsePart> {
+                for await (const chunk of stream) {
+                    const content = chunk.token.text.replace(/<\|endoftext\|>/g, '');
```

Comment:
Will this work for most/all hugging face models or only for starcoder? If this is starcoder specific then we should check for that model before setting the parameters / replacing the content.

Reply:
I found that almost all HuggingFace Model return stop charakters. And that it is most robust to explcitly specify it and then replace it. Good test case is meta-llama/Llama-3.2-3B-Instruct which is always "warm"